### PR TITLE
Add Storage Account Auth Via SAS Token

### PR DESCRIPTION
**Pull request by jspearman3**
Hey, my team has been using this library to capture events within our azure storage table, however, we are moving away from using the storage account key as an authorization procedure. Instead, we are switching over to using SAS tokens, and it would be great if we could continue using this library.

I was able to fork the repo and create a version of the library that appeared to work for us. This PR contains the changes I was working with (not a huge change). Is the idea of adding SAS token auth support an idea you'd be able to get behind? And if so, is this implementation appropriate?

Thanks.

### CODE REVIEW

File: src/CosmoStore.TableStorage/EventStore.fs [hunk 1]
```diff
@@ -147,6 +147,9 @@ let getEventStore (configuration:Configuration) =
         | Cloud (accountName, authKey) -> 
             let credentials = Auth.StorageCredentials(accountName, authKey)
             CloudStorageAccount(credentials, true)
+        | CloudBySAS (accountName, sasToken) ->
+            let credentials = Auth.StorageCredentials(sasToken)
+            CloudStorageAccount(credentials, accountName, "core.windows.net", true )
```
Comment (Dzoukr) [inline src/CosmoStore.TableStorage/EventStore.fs:152]: Why not just `CloudStorageAccount(credentials, true)` like is on line 149?

**Comment by Dzoukr** [comment]
Hi @jspearman3, look really good. Please check my comment on creating `CloudStorageAccount` for SAS. Also I am thinking about this:

```fsharp
type CloudAuth =
    | AuthKey of string
    | SASToken of string

type StorageAccount =
    | Cloud of CloudAuth
    | LocalEmulator
```

but it would be kind of breaking change. 🤔 What do you think?

**Comment by jspearman3** [comment]
I'd prefer that typing change as far as domain design goes for sure. But yeah, I figured adding a new case to the existing DU would be less of a breakage, so I stuck with that. To us it wouldn't matter as we would be upgrading the package anyway.

Also, for the comment you had, there needs to be a place to specify the account name, and that is the only overload that does. Else you get the following exception:

```
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'AccountName')
   at Microsoft.WindowsAzure.Storage.Core.Util.CommonUtility.AssertNotNull(String paramName, Object value)
   at Microsoft.WindowsAzure.Storage.CloudStorageAccount..ctor(StorageCredentials storageCredentials, String accountName, String endpointSuffix, Boolean useHttps)
   at Microsoft.WindowsAzure.Storage.CloudStorageAccount..ctor(StorageCredentials storageCredentials, String endpointSuffix, Boolean useHttps)
   at Microsoft.WindowsAzure.Storage.CloudStorageAccount..ctor(StorageCredentials storageCredentials, Boolean useHttps)
   at Program.main(String[] argv) in /home/jimmy/Projects/dotnet/SANDBOX/tokenfetch/Program.fs:line 381
```

**Comment by Dzoukr** [comment]
Published as `v3.1.0` - thanks for that!

**Comment by Dzoukr** [comment]
@ninjarobot You probable meant this package? https://www.nuget.org/packages/CosmoStore.TableStorage/3.1.0

**Pull request merged**